### PR TITLE
Switch to terraform 0.12

### PR DESCRIPTION
**Pull request by corest**
Towards https://github.com/giantswarm/giantswarm/issues/3757

### CODE REVIEW

**Comment by corest** [comment]
Tested. Works fine

File: Dockerfile [hunk 1]
```diff
@@ -19,36 +18,30 @@ RUN echo "deb [arch=amd64] https://packages.microsoft.com/repos/azure-cli/ wheez
...
-
-# We need to build ct_config terraform provider from sources, because of two things:
-# 1. it's still not avaialble in terraform repos [1].
-# 2. latest binary release v0.2.1 is too old and does not support passwd groups.
-#
-# Links:
-# 1 - https://github.com/coreos/terraform-provider-ct/issues/21
-
-# Install Go 1.10.
+# Install Go 1.11.
```
Comment (tuommaki) [inline Dockerfile:26]: No Go 1.12 available yet? Current stable upstream is `1.12.5`.

**Review by tuommaki — APPROVED** [review]
Added note about possible upgrade in Go. Otherwise seems good to me when it works.

File: Dockerfile (hunk 1 above)
Comment (corest) [reply Dockerfile:26]: I had some issues during builds with go 1.12, so set to 1.11. Tested with that version locally. So for the current branch of gotemplate provider it is fine

**Pull request merged**